feature pooling-allocator doesn't compile on its own

Trying to get parts of wasmtime to compile without enlisting all features.

The pooling-allocator feature pulls in some code but uses a field that is only defined when the crate/wasmtime feature "async" is defined.

```
wasmtime $ cargo check --no-default-features --features pooling-allocator
error[E0599]: no method named `total_stacks` found for struct `PoolingAllocationConfig` in the current scope
   --> crates/cli-flags/src/lib.rs:593:29
    |
593 |                         cfg.total_stacks(limit);
    |                             ^^^^^^^^^^^^
    |
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -369,8 +369,10 @@ jobs:
           - name: wasmtime-cli
             checks: |
               -p wasmtime-cli --no-default-features
+              -p wasmtime-cli --no-default-features --features pooling-allocator
               -p wasmtime-cli --no-default-features --features run
               -p wasmtime-cli --no-default-features --features run,component-model
+              -p wasmtime-cli --no-default-features --features run,pooling-allocator
               -p wasmtime-cli --no-default-features --features compile
               -p wasmtime-cli --no-default-features --features compile,cranelift
               -p wasmtime-cli --no-default-features --features compile,cranelift,component-model

--- a/crates/cli-flags/Cargo.toml
+++ b/crates/cli-flags/Cargo.toml
@@ -21,6 +21,7 @@ wasmtime = { workspace = true, features = ["gc"] }
 humantime = { workspace = true }
 
 [features]
+async = ["wasmtime/async"]
 pooling-allocator = ["wasmtime/pooling-allocator"]
 component-model = ["wasmtime/component-model"]
 cache = ["wasmtime/cache"]

--- a/crates/cli-flags/src/lib.rs
+++ b/crates/cli-flags/src/lib.rs
@@ -589,8 +589,10 @@ impl CommonOptions {
                     if let Some(limit) = self.opts.pooling_total_tables {
                         cfg.total_tables(limit);
                     }
-                    if let Some(limit) = self.opts.pooling_total_stacks {
-                        cfg.total_stacks(limit);
+                    match_feature! {
+                        ["async" : self.opts.pooling_total_stacks]
+                        limit => cfg.total_stacks(limit),
+                        _ => err,
                     }
                     if let Some(limit) = self.opts.pooling_max_memory_size {
                         cfg.max_memory_size(limit);

